tests/test_recompile_ux.py

```diff
@@ -0,0 +1,92 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import Any
+from typing import List
+
+import torch
+
+import torchdynamo
+import torchdynamo.config
+import torchdynamo.testing
+
+torchdynamo.config.debug = False
+from torchdynamo.utils import counters
+
+class RecompileUxTests(torchdynamo.testing.TestCase):
+    # TODO(whc) dynamo actualy recompiles one more time than the cache limit
+    max_recompiles = 2
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        torchdynamo.config.cache_size_limit = max(1, RecompileUxTests.max_recompiles - 1)
+
+    def test_loop_torture(self):
+        def loop_torture(input):
+            out = input
+            for i in range(torch.randint(low=0, high=1000, size=())):
+                out += input
+            return out
+
+        for i in range(10):
+            x = torch.randn(3)
+            with torchdynamo.optimize("eager"):
+                out = loop_torture(x)
+
+        # Currently, we recompile each time,
+        # We'd probably like to bail out quickly and warn
+        self.assertEqual(counters["frames"]["total"], RecompileUxTests.max_recompiles)
+        self.assertEqual(
+            torchdynamo.utils.counters["frames"]["total"],
+            torchdynamo.utils.counters["frames"]["ok"],
+        )
+
+    def test_dynamic_input(self):
+        def model(input):
+            return input + input
+
+        for i in range(10):
+            bsz = torch.randint(low=0, high=1000, size=())
+            x = torch.randn((bsz, 3, 4))
+            with torchdynamo.optimize("eager"):
+                out = model(x)
+        
+        print(counters)
+        self.assertEqual(counters["frames"]["total"], TortureTests.max_recompiles)
+        self.assertEqual(
+            torchdynamo.utils.counters["frames"]["total"],
+            torchdynamo.utils.counters["frames"]["ok"],
+        )
+
+    def test_nvfuser_guards(self):
```

Comment:
Likely need a requires cuda `unittest.skipIf`.   We have a that in some other files that you could refector into a common place.